lighthouse-cli/bin.ts

```diff
@@ -276,38 +276,36 @@ function saveResults(results: Results,
     return promise.then(_ => Printer.write(results, flags.output, flags.outputPath));
 }
 
-export function runLighthouse(url: string,
+export async function runLighthouse(url: string,
                        flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
                          outputPath: string, interactive: boolean, saveArtifacts: boolean, saveAssets: boolean
                          maxWaitForLoad: number},
-                       config: Object | null): Promise<undefined> {
-
-  let chromeLauncher: ChromeLauncher;
-  return initPort(flags)
-    .then(() => getDebuggableChrome(flags))
-    .then(chrome => chromeLauncher = chrome)
-    .then(() => lighthouse(url, flags, config))
-    .then((results: Results) => {
-      // remove artifacts from result so reports won't include artifacts.
-      const artifacts = results.artifacts;
-      delete results.artifacts;
-
-      let promise = saveResults(results, artifacts!, flags);
-      if (flags.interactive) {
-        promise = promise.then(() => performanceXServer.hostExperiment({url, flags, config}, results));
-      }
+                       config: Object | null): Promise<{}|void> {
 
-      return promise;
-    })
-    .then(() => chromeLauncher.kill())
-    .catch(err => {
-      if (typeof chromeLauncher !== 'undefined') {
-        return chromeLauncher.kill().then(() => handleError(err), handleError);
-      }
+  let chromeLauncher: ChromeLauncher | undefined = undefined;
 
-      return handleError(err);
-    })
-}
+  try {
+    await initPort(flags)
+    const chromeLauncher = await getDebuggableChrome(flags)
+    const results = await lighthouse(url, flags, config);
+
+    const artifacts = results.artifacts;
+    delete results.artifacts;
+
+    await saveResults(results, artifacts!, flags);
+    if (flags.interactive) {
+      await performanceXServer.hostExperiment({url, flags, config}, results);
+    }
+
+    return await chromeLauncher.kill();
+  } catch (err) {
+    if (typeof chromeLauncher !== 'undefined') {
+      await chromeLauncher!.kill();
```

Comment:
Non-null assertion operator
---

A new ! post-fix expression operator may be used to assert that its operand is non-null and non-undefined in contexts where the type checker is unable to conclude that fact. Specifically, the operation x! produces a value of the type of x with null and undefined excluded. Similar to type assertions of the forms <T>x and x as T, the ! non-null assertion operator is simply removed in the emitted JavaScript code.